### PR TITLE
track2 .NET SDK generation for dataprotection

### PR DESCRIPTION
**Pull request by asmitt**
# Contributing to the Azure SDK

Please see our [CONTRIBUTING.md](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md) if you are not familiar with contributing to this repository or have questions.

For specific information about pull request etiquette and best practices, see [this section](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#pull-request-etiquette-and-best-practices).

### CODE REVIEW

File: sdk/dataprotection/Azure.ResourceManager.DataProtectionBackup/api/Azure.ResourceManager.DataProtectionBackup.netstandard2.0.cs [hunk 1]
```diff
@@ -900,6 +922,31 @@ public partial class CopyOnExpirySetting : Azure.ResourceManager.DataProtectionB
     {
         public CopyOnExpirySetting() { }
     }
+    public partial class CrossRegionRestoreDetails
+    {
+        public CrossRegionRestoreDetails(string sourceRegion, string sourceBackupInstanceId) { }
+        public string SourceBackupInstanceId { get { throw null; } }
```
Comment (ArcturusZhang) [inline sdk/dataprotection/Azure.ResourceManager.DataProtectionBackup/api/Azure.ResourceManager.DataProtectionBackup.netstandard2.0.cs:928]: this looks like an arm resource. Could you please confirm if this is always an arm resource id?
If so, please change the format of this property to `arm-id`

File: sdk/dataprotection/Azure.ResourceManager.DataProtectionBackup/api/Azure.ResourceManager.DataProtectionBackup.netstandard2.0.cs [hunk 2]
```diff
@@ -900,6 +922,31 @@ public partial class CopyOnExpirySetting : Azure.ResourceManager.DataProtectionB
...
+        public string SourceRegion { get { throw null; } }
+    }
+    public partial class CrossRegionRestoreJobContent
+    {
+        public CrossRegionRestoreJobContent(string sourceRegion, string sourceBackupVaultId, string jobId) { }
+        public string JobId { get { throw null; } }
+        public string SourceBackupVaultId { get { throw null; } }
```
Comment (ArcturusZhang) [inline sdk/dataprotection/Azure.ResourceManager.DataProtectionBackup/api/Azure.ResourceManager.DataProtectionBackup.netstandard2.0.cs:935]: same here, are these ids `arm resource id`? If so please change their format to `arm-id` accordingly.

File: sdk/dataprotection/Azure.ResourceManager.DataProtectionBackup/api/Azure.ResourceManager.DataProtectionBackup.netstandard2.0.cs [hunk 3]
```diff
@@ -900,6 +922,31 @@ public partial class CopyOnExpirySetting : Azure.ResourceManager.DataProtectionB
...
+        public string SourceRegion { get { throw null; } }
```
Comment (ArcturusZhang) [inline sdk/dataprotection/Azure.ResourceManager.DataProtectionBackup/api/Azure.ResourceManager.DataProtectionBackup.netstandard2.0.cs:936]: same here, is this an azure location?

File: sdk/dataprotection/Azure.ResourceManager.DataProtectionBackup/api/Azure.ResourceManager.DataProtectionBackup.netstandard2.0.cs [hunk 4]
```diff
@@ -900,6 +922,31 @@ public partial class CopyOnExpirySetting : Azure.ResourceManager.DataProtectionB
...
+    }
+    public partial class CrossRegionRestoreJobsContent
+    {
+        public CrossRegionRestoreJobsContent(string sourceRegion, string sourceBackupVaultId) { }
+        public string SourceBackupVaultId { get { throw null; } }
+        public string SourceRegion { get { throw null; } }
```
Comment (ArcturusZhang) [inline sdk/dataprotection/Azure.ResourceManager.DataProtectionBackup/api/Azure.ResourceManager.DataProtectionBackup.netstandard2.0.cs:942]: same questions here

File: sdk/dataprotection/Azure.ResourceManager.DataProtectionBackup/api/Azure.ResourceManager.DataProtectionBackup.netstandard2.0.cs [hunk 5]
```diff
@@ -1237,6 +1285,7 @@ public partial class DataProtectionBackupVaultProperties
         public Azure.ResourceManager.DataProtectionBackup.Models.BackupVaultFeatureSettings FeatureSettings { get { throw null; } set { } }
         public bool? IsVaultProtectedByResourceGuard { get { throw null; } set { } }
         public Azure.ResourceManager.DataProtectionBackup.Models.DataProtectionBackupProvisioningState? ProvisioningState { get { throw null; } }
+        public System.Collections.Generic.IList<string> ReplicatedRegions { get { throw null; } }
```
Comment (ArcturusZhang) [inline sdk/dataprotection/Azure.ResourceManager.DataProtectionBackup/api/Azure.ResourceManager.DataProtectionBackup.netstandard2.0.cs:1288]: are the items here `azure locations`? if so we could change their format to `azure-location`

File: sdk/dataprotection/Azure.ResourceManager.DataProtectionBackup/api/Azure.ResourceManager.DataProtectionBackup.netstandard2.0.cs [hunk 6]
```diff
@@ -1395,6 +1444,12 @@ public partial class DeletedDataProtectionBackupInstanceProperties : Azure.Resou
         public static bool operator !=(Azure.ResourceManager.DataProtectionBackup.Models.FeatureSupportStatus left, Azure.ResourceManager.DataProtectionBackup.Models.FeatureSupportStatus right) { throw null; }
         public override string ToString() { throw null; }
     }
+    public partial class FetchSecondaryRPsRequestContent
+    {
+        public FetchSecondaryRPsRequestContent() { }
+        public string SourceBackupInstanceId { get { throw null; } set { } }
+        public string SourceRegion { get { throw null; } set { } }
```
Comment (ArcturusZhang) [inline sdk/dataprotection/Azure.ResourceManager.DataProtectionBackup/api/Azure.ResourceManager.DataProtectionBackup.netstandard2.0.cs:1451]: same question here

File: sdk/dataprotection/Azure.ResourceManager.DataProtectionBackup/api/Azure.ResourceManager.DataProtectionBackup.netstandard2.0.cs [hunk 7]
```diff
@@ -1462,6 +1517,22 @@ public partial class KubernetesClusterRestoreCriteria : Azure.ResourceManager.Da
...
+        public System.Collections.Generic.IList<string> ExcludedResourceTypes { get { throw null; } }
+        public bool IncludeClusterScopeResources { get { throw null; } }
+        public System.Collections.Generic.IList<string> IncludedNamespaces { get { throw null; } }
+        public System.Collections.Generic.IList<string> IncludedResourceTypes { get { throw null; } }
+        public System.Collections.Generic.IList<string> LabelSelectors { get { throw null; } }
+        public System.Collections.Generic.IDictionary<string, string> NamespaceMappings { get { throw null; } }
+        public Azure.ResourceManager.DataProtectionBackup.Models.PersistentVolumeRestoreMode? PersistentVolumeRestoreMode { get { throw null; } set { } }
+        public System.Collections.Generic.IList<Azure.ResourceManager.DataProtectionBackup.Models.NamespacedName> RestoreHookReferences { get { throw null; } }
+        public Azure.Core.ResourceIdentifier StagingResourceGroupId { get { throw null; } set { } }
+        public Azure.Core.ResourceIdentifier StagingStorageAccountId { get { throw null; } set { } }
```
Comment (ArcturusZhang) [inline sdk/dataprotection/Azure.ResourceManager.DataProtectionBackup/api/Azure.ResourceManager.DataProtectionBackup.netstandard2.0.cs:1534]: are these arm resource ids?

File: sdk/dataprotection/Azure.ResourceManager.DataProtectionBackup/api/Azure.ResourceManager.DataProtectionBackup.netstandard2.0.cs [hunk 8]
```diff
@@ -1509,6 +1580,24 @@ public partial class RangeBasedItemLevelRestoreCriteria : Azure.ResourceManager.
         public string MaxMatchingValue { get { throw null; } set { } }
         public string MinMatchingValue { get { throw null; } set { } }
     }
+    [System.Runtime.InteropServices.StructLayoutAttribute(System.Runtime.InteropServices.LayoutKind.Sequential)]
+    public readonly partial struct RecoveryPointCompletionState : System.IEquatable<Azure.ResourceManager.DataProtectionBackup.Models.RecoveryPointCompletionState>
```
Comment (ArcturusZhang) [inline sdk/dataprotection/Azure.ResourceManager.DataProtectionBackup/api/Azure.ResourceManager.DataProtectionBackup.netstandard2.0.cs:1584]: this name is quite generic and a bit likely to collide with types in recovery service.
Could you please add some prefixes or suffixes to make this more specific to data protection?

**Review by ArcturusZhang — COMMENTED** [review]
A few naming comments.
Here is a document on how to change them: https://github.com/Azure/autorest.csharp/blob/feature/v3/docs/mgmt/polishing.md#rename-mapping
Please note the name changing following above document will only change their "display name" in the SDK library, their name in the json payload will never change.

File: sdk/dataprotection/Azure.ResourceManager.DataProtectionBackup/api/Azure.ResourceManager.DataProtectionBackup.netstandard2.0.cs [hunk 9]
```diff
@@ -470,11 +490,11 @@ public static partial class ArmDataProtectionBackupModelFactory
...
+        public static Azure.ResourceManager.DataProtectionBackup.Models.BackupJobExtendedInfo BackupJobExtendedInfo(System.Collections.Generic.IReadOnlyDictionary<string, string> additionalDetails = null, string backupInstanceState = null, double? dataTransferredInBytes = default(double?), string recoveryDestination = null, Azure.ResourceManager.DataProtectionBackup.Models.RestoreJobRecoveryPointDetails sourceRecoverPoint = null, System.Collections.Generic.IEnumerable<Azure.ResourceManager.DataProtectionBackup.Models.BackupJobSubTask> subTasks = null, Azure.ResourceManager.DataProtectionBackup.Models.RestoreJobRecoveryPointDetails targetRecoverPoint = null, System.Collections.Generic.IEnumerable<Azure.ResourceManager.DataProtectionBackup.Models.UserFacingWarningDetail> warningDetails = null) { throw null; }
```
Comment (ArcturusZhang) [inline sdk/dataprotection/Azure.ResourceManager.DataProtectionBackup/api/Azure.ResourceManager.DataProtectionBackup.netstandard2.0.cs:493]: ApiCompat is reporting breaking changes for the 4 methods changed in the model factory.
Could you please add the previous version of these 4 methods into the partial class of the model factory in the `Customized` directory?
Their implementations could be simply calling their generated variant.

**Review by jsquire — DISMISSED** [review]
@asmitt:

Our automation is identifying you as a community contributor.  Please review the Azure SDK onboarding documentation and ensure that your GitHub organization membership and permissions are correct.
- [Azure SDK onboarding](https://eng.ms/docs/products/azure-developer-experience/onboard/requirements) _(Microsoft internal)_
- [Azure SDK onboarding assistance](https://teams.microsoft.com/l/channel/19%3ade3fce22c2994be18cac50502c55f717%40thread.skype/Azure%2520SDK%2520Onboarding?groupId=3e17dcb0-4257-4a30-b843-77f47f1d4121&tenantId=72f988bf-86f1-41af-91ab-2d7cd011db47) _(Microsoft internal)_

Please also be sure to add yourself to [CODEOWNERS](https://github.com/Azure/azure-sdk-for-net/blob/main/.github/CODEOWNERS) for this library, if you will be maintaining it going forward.

File: sdk/dataprotection/Azure.ResourceManager.DataProtectionBackup/api/Azure.ResourceManager.DataProtectionBackup.netstandard2.0.cs (hunk 7 above)
Comment (asmitt) [reply sdk/dataprotection/Azure.ResourceManager.DataProtectionBackup/api/Azure.ResourceManager.DataProtectionBackup.netstandard2.0.cs:1534]: yes

File: sdk/dataprotection/Azure.ResourceManager.DataProtectionBackup/api/Azure.ResourceManager.DataProtectionBackup.netstandard2.0.cs (hunk 3 above)
Comment (asmitt) [reply sdk/dataprotection/Azure.ResourceManager.DataProtectionBackup/api/Azure.ResourceManager.DataProtectionBackup.netstandard2.0.cs]: added the correct formats

File: sdk/dataprotection/Azure.ResourceManager.DataProtectionBackup/api/Azure.ResourceManager.DataProtectionBackup.netstandard2.0.cs (hunk 5 above)
Comment (asmitt) [reply sdk/dataprotection/Azure.ResourceManager.DataProtectionBackup/api/Azure.ResourceManager.DataProtectionBackup.netstandard2.0.cs]: added the correct formats

File: sdk/dataprotection/Azure.ResourceManager.DataProtectionBackup/api/Azure.ResourceManager.DataProtectionBackup.netstandard2.0.cs (hunk 6 above)
Comment (asmitt) [reply sdk/dataprotection/Azure.ResourceManager.DataProtectionBackup/api/Azure.ResourceManager.DataProtectionBackup.netstandard2.0.cs]: added the correct formats

**Comment by asmitt** [comment]
@ArcturusZhang We had to change format for one of the existing field, "DataProtectionOperationJobExtendedInfo.JobResourceId", to string from arm-id since it can also be uuid in some new scenarios. But this is ApiCompat issue. How can we resolve these?

**Review by ArcturusZhang — APPROVED** [review]

**Pull request merged**